### PR TITLE
[Keyboard] Add Wombat

### PR DESCRIPTION
**Pull request by Xelus22**
## Description

Add Wombat

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xelus/wombat/info.json [hunk 1]
```diff
@@ -0,0 +1,306 @@
+{
+    "manufacturer": "Xelus",
+    "keyboard_name": "Wombat",
+    "maintainer": "Xelus22",
+    "bootloader": "stm32-dfu",
+    "bootmagic": {
+        "enabled": true
+    },
+    "debounce": 5,
```
Comment (zvecr) [inline keyboards/xelus/wombat/info.json:9]:
```suggestion
```

File: keyboards/xelus/wombat/info.json [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+            {"pin_a": "B7", "pin_b": "B8", "resolution": 4},
+            {"pin_a": "B9", "pin_b": "E0", "resolution": 4},
+            {"pin_a": "E1", "pin_b": "E2", "resolution": 4},
+            {"pin_a": "E3", "pin_b": "E4", "resolution": 4},
+            {"pin_a": "E5", "pin_b": "E6", "resolution": 4},
+            {"pin_a": "C13", "pin_b": "C14", "resolution": 4},
+            {"pin_a": "C15", "pin_b": "H0", "resolution": 4},
+            {"pin_a": "H1", "pin_b": "C0", "resolution": 4},
+            {"pin_a": "C1", "pin_b": "C2", "resolution": 4},
+            {"pin_a": "C3", "pin_b": "A0", "resolution": 4}
```
Comment (zvecr) [inline keyboards/xelus/wombat/info.json:39]:
```suggestion
            {"pin_a": "B11", "pin_b": "B12"},
            {"pin_a": "B13", "pin_b": "B14"},
            {"pin_a": "B15", "pin_b": "D8"},
            {"pin_a": "D9", "pin_b": "D10"},
            {"pin_a": "D11", "pin_b": "D12"},
            {"pin_a": "D13", "pin_b": "D14"},
            {"pin_a": "D15", "pin_b": "C6"},
            {"pin_a": "C7", "pin_b": "C8"},
            {"pin_a": "C9", "pin_b": "A8"},
            {"pin_a": "A9", "pin_b": "A10"},
            {"pin_a": "B7", "pin_b": "B8"},
            {"pin_a": "B9", "pin_b": "E0"},
            {"pin_a": "E1", "pin_b": "E2"},
            {"pin_a": "E3", "pin_b": "E4"},
            {"pin_a": "E5", "pin_b": "E6"},
            {"pin_a": "C13", "pin_b": "C14"},
            {"pin_a": "C15", "pin_b": "H0"},
            {"pin_a": "H1", "pin_b": "C0"},
            {"pin_a": "C1", "pin_b": "C2"},
            {"pin_a": "C3", "pin_b": "A0"}
```

File: keyboards/xelus/wombat/info.json [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+    "features": {
+        "audio": false,
+        "bootmagic": true,
+        "command": true,
+        "console": true,
+        "encoder": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": false
```
Comment (zvecr) [inline keyboards/xelus/wombat/info.json:51]:
```suggestion
        "bootmagic": true,
        "command": false,
        "console": false,
        "encoder": true,
        "extrakey": true,
        "mousekey": true,
        "nkro": true
```

File: keyboards/xelus/wombat/info.json [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+    },
+    "matrix_pins": {
+        "cols": ["C11", "C12", "D0", "D1", "B10", "E15", "E14", "E13", "E12", "E11", "A13", "A1", "C5", "C4"],
+        "rows": ["B6", "B5", "B3", "B4", "D6", "D7", "D4", "D5", "D2", "D3", "A15", "C10", "A3", "A2", "B1", "B0", "B2", "E7"]
+    },
+    "mouse_key": {
+        "enabled": true
+    },
```
Comment (zvecr) [inline keyboards/xelus/wombat/info.json:59]:
```suggestion
```

File: keyboards/xelus/wombat/info.json [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+    "processor": "STM32L433",
+    "url": "https://github.com/qmk/qmk_firmware/tree/master/keyboards/xelus/wombat",
+    "usb": {
+        "device_version": "0.0.1",
+        "force_nkro": false,
```
Comment (zvecr) [inline keyboards/xelus/wombat/info.json:64]:
```suggestion
```

File: keyboards/xelus/wombat/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                                 KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                    KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                    KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)
+};
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/xelus/wombat/keymaps/default/keymap.c:48]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/xelus/wombat/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                                 KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                    KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                    KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)
+};
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/xelus/wombat/keymaps/via/keymap.c:74]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/xelus/wombat/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+              ENCODER_CCW_CW(KC_TRNS, KC_TRNS),
+              ENCODER_CCW_CW(KC_TRNS, KC_TRNS),
+              ENCODER_CCW_CW(KC_TRNS, KC_TRNS),
+              ENCODER_CCW_CW(KC_TRNS, KC_TRNS),
+              ENCODER_CCW_CW(KC_TRNS, KC_TRNS),
+              ENCODER_CCW_CW(KC_TRNS, KC_TRNS),
+              ENCODER_CCW_CW(KC_TRNS, KC_TRNS),
+              ENCODER_CCW_CW(KC_TRNS, KC_TRNS),
+              ENCODER_CCW_CW(KC_TRNS, KC_TRNS)
+            }
```
Comment (zvecr) [inline keyboards/xelus/wombat/keymaps/via/keymap.c:158]:
```suggestion
            }
```

File: keyboards/xelus/wombat/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                                 KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                    KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                    KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)
```
Comment (zvecr) [inline keyboards/xelus/wombat/keymaps/via/keymap.c:70]:
```suggestion
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                    KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)
```

File: keyboards/xelus/wombat/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+            }
+            break;
+        case 19:
+            if (clockwise) {
+                tap_code(KC_PGDN);
+            } else {
+                tap_code(KC_PGUP);
+            }
+            break;
+    }
```
Comment (zvecr) [inline keyboards/xelus/wombat/keymaps/via/keymap.c:303]: This doesnt have much value over
```suggestion
        if (clockwise) {
            tap_code(KC_PGDN);
        } else {
            tap_code(KC_PGUP);
        }
```

File: keyboards/xelus/wombat/rules.mk [hunk 11]
```diff
@@ -0,0 +1,3 @@
+# Intentionally left blank
+
```
Comment (zvecr) [inline keyboards/xelus/wombat/rules.mk:2]: File isnt blank...
```suggestion
```

File: keyboards/xelus/wombat/readme.md [hunk 12]
```diff
@@ -0,0 +1,13 @@
+# Wombat
```
Comment (zvecr) [inline keyboards/xelus/wombat/readme.md:1]: Does not follow [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md)

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/xelus/wombat/info.json [hunk 13]
```diff
@@ -0,0 +1,299 @@
+{
+    "manufacturer": "Xelus",
+    "keyboard_name": "Wombat",
+    "maintainer": "Xelus22",
+    "bootloader": "stm32-dfu",
+    "bootmagic": {
+        "enabled": true
+    },
```
Comment (waffle87) [inline keyboards/xelus/wombat/info.json:8]:
```suggestion
```

File: keyboards/xelus/wombat/info.json [hunk 14]
```diff
@@ -0,0 +1,299 @@
...
+    "diode_direction": "COL2ROW",
+    "dynamic_keymap": {
+        "eeprom_max_addr": 32767
+    },
+    "eeprom": {
+        "driver": "wear_leveling"
+    },
+    "encoder": {
+        "enabled": true,
```
Comment (waffle87) [inline keyboards/xelus/wombat/info.json:17]:
```suggestion
```

File: keyboards/xelus/wombat/config.h [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+// SPI EEPROM left in because I couldn't get it working
```
Comment (drashna) [inline keyboards/xelus/wombat/config.h:19]: Odd. I'm curious which chip you're using.

File: keyboards/xelus/wombat/info.json [hunk 16]
```diff
@@ -0,0 +1,295 @@
+{
+    "manufacturer": "Xelus",
+    "keyboard_name": "Wombat",
+    "maintainer": "Xelus22",
+    "bootloader": "stm32-dfu",
+    "diode_direction": "COL2ROW",
+    "dynamic_keymap": {
+        "eeprom_max_addr": 32767
+    },
```
Comment (drashna) [inline keyboards/xelus/wombat/info.json:9]: This shouldn't be needed, should be set to whatever the wear leveling size is.  Eg, this should only be needed if you want/need to have a different size.

File: keyboards/xelus/wombat/config.h (hunk 15 above)
Comment (Xelus22) [reply keyboards/xelus/wombat/config.h:19]: 25CSM04

**Review by drashna — APPROVED** [review]

File: keyboards/xelus/wombat/wombat.c [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
```
Comment (fauxpark) [inline keyboards/xelus/wombat/wombat.c:15]:
```suggestion
 */

#include "quantum.h"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**